File: Java/java_ai_generated_codes/src/MediumLevel/LlaMa/PrimorialCalculator.java
package MediumLevel.LlaMa;

public class PrimorialCalculator {
    public static long primorial(int n) {
        int count = 0;
        long result = 1;
        int i = 2;
        while (count < n) {
            if (isPrime(i)) {
                result *= i;
                count++;
            }
            i++;
        }
        return result;
    }

    public static boolean isPrime(int n) {
        if (n <= 1) {
            return false;
        }
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: Scala/scala_ai_generated_codes/src/main/scala/MediumLevel/LlaMa/PrimorialCalculator.scala
package MediumLevel.LlaMa

object PrimorialCalculator {
  def isPrime(n: Int): Boolean = {
    if (n <= 1) false
    else if (n == 2) true
    else !(2 until n).exists(x => n % x == 0)
  }

  def primorial(n: Int): BigInt = {
    var result = BigInt(1)
    var count = 0
    var i = 2
    while (count < n) {
      if (isPrime(i)) {
        result *= i
        count += 1
      }
      i += 1
    }
    result
  }

  def main(args: Array[String]): Unit = {
    val n = 3
    val prim = primorial(n)
    println(s"The primorial of $n is $prim")
  }
}
